File: book_rental_server/src/main/java/com/bookrental/service/BookService.java
package com.bookrental.service;

import java.util.List;

import com.bookrental.model.Book;

public interface BookService {

	//Add Book
	public Book addBook(Book book);
	
	//Get All Books 
	public List<Book> getAllBook();
	
	//Get specific Book using id
	public Book getBook(Long id);
	
	//Update specific book using id
	public Book updateBook(Book book, Long id);
	
	//Delete Specific Using ID
	public void deleteBook(Long id);
}

File: book_rental_frontend/src/app/service/book.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import base_url from './base_url';

@Injectable({
  providedIn: 'root'
})
export class BookService {

  constructor(private _http:HttpClient) { }

  //get all books
  public getAllBooks(){
    return this._http.get(`${base_url}/book`);
  }

  //add books
  public addBook(book:any){
    return this._http.post(`${base_url}/book`, book);
  }

  //Get one book
  public getBook(bookId:any){
    return this._http.get(`${base_url}/book/${bookId}`);
  }

  //update book
  public updateBook(book:any, bookId:any){
    return this._http.put(`${base_url}/book/${bookId}`, book);
  }

  //delete book
  public deleteBook(bookId:any){
    return this._http.delete(`${base_url}/book/${bookId}`);
  }
}
